pkg/flow/internallogger/logs.go

```diff
@@ -0,0 +1,211 @@
+	logger.sendToWorker(recipientID, tags, util.Debug, msg)
+}
+
+func (logger *Logger) Debugf(ctx context.Context, recipientID uuid.UUID, tags map[string]string, msg string, a ...interface{}) {
+	logger.Telemetry(ctx, util.Debug, tags, msg)
+	logger.sendToWorker(recipientID, tags, util.Debug, fmt.Sprintf(msg, a...))
+}
+
+func (logger *Logger) Info(ctx context.Context, recipientID uuid.UUID, tags map[string]string, msg string) {
+	logger.Telemetry(ctx, util.Info, tags, msg)
+	logger.sendToWorker(recipientID, tags, util.Info, msg)
+}
+
+func (logger *Logger) Infof(ctx context.Context, recipientID uuid.UUID, tags map[string]string, msg string, a ...interface{}) {
+	logger.Telemetry(ctx, util.Info, tags, msg)
+	logger.sendToWorker(recipientID, tags, util.Info, fmt.Sprintf(msg, a...))
+}
+
+func (logger *Logger) Error(ctx context.Context, recipientID uuid.UUID, tags map[string]string, msg string) {
+	logger.Telemetry(ctx, util.Error, tags, msg)
+	logger.sendToWorker(recipientID, tags, util.Error, msg)
+}
+
+func (logger *Logger) Errorf(ctx context.Context, recipientID uuid.UUID, tags map[string]string, msg string, a ...interface{}) {
+	logger.Telemetry(ctx, util.Error, tags, msg)
+	logger.sendToWorker(recipientID, tags, util.Error, fmt.Sprintf(msg, a...))
+}
+
+func (logger *Logger) Panic(ctx context.Context, recipientID uuid.UUID, tags map[string]string, msg string) {
+	logger.Telemetry(ctx, util.Panic, tags, msg)
+	logger.sendToWorker(recipientID, tags, util.Panic, msg)
+}
+
+func (logger *Logger) Panicf(ctx context.Context, recipientID uuid.UUID, tags map[string]string, msg string, a ...interface{}) {
+	logger.Telemetry(ctx, util.Panic, tags, msg)
+	logger.sendToWorker(recipientID, tags, util.Panic, fmt.Sprintf(msg, a...))
+}
+
+func (logger *Logger) sendToWorker(recipientID uuid.UUID, tags map[string]string, level string, msg string) {
+	defer func() {
+		_ = recover()
+	}()
+
+	logger.logQueue <- &logMessage{
+		t:           time.Now(),
+		msg:         msg,
+		tags:        tags,
+		level:       level,
+		recipientID: recipientID,
+	}
+}
+
+func (logger *Logger) Telemetry(ctx context.Context, level string, tags map[string]string, msg string) {
+	span := trace.SpanFromContext(ctx)
+	tid := span.SpanContext().TraceID()
+	if len(tags) == 0 {
+		logger.sugar.Infow(msg, "trace", tid)
+	} else {
+		ar := make([]interface{}, len(tags)*2)
+		i := 0
+		for k, v := range tags {
+			ar[i] = k
+			ar[i+1] = v
+			i = i + 2
+		}
+		ar = append(ar, "trace", tid)
+		switch level {
+		case util.Info:
+			logger.sugar.Infow(msg, ar...)
+		case util.Debug:
+			logger.sugar.Debugw(msg, ar...)
+		case util.Error:
+			logger.sugar.Errorw(msg, ar...)
+		case util.Panic:
+			logger.sugar.Panicw(msg, ar...)
+		default:
+			logger.sugar.Debugw(msg, ar...) // this should never happen
+		}
+	}
+}
+
+func (logger *Logger) SendLogMsgToDB(l *logMessage) error {
+	ctx := context.Background() // logs are often queued and stored after their originating requests have ended.
```

Comment:
This is true. This may need further testing/investigation to properly interact with tracing.